src/sec_certs/dataset/auxiliary_dataset_handling.py

```diff
@@ -30,8 +29,8 @@ class AuxiliaryDatasetHandler(ABC):
     aux_datasets_dir: Path
```

Comment:
Just set this to `Path |  None` to avoid the type warnings?

Reply:
Well, we need to lie here as well I think. The same way we are lying in the `Dataset.root_dir` property claiming that it is never `None` and only protecting it via the `only_backed` decorator. Otherwise, typewise it just all goes wrong and every access to any attribute derived from `root_dir` needs to be checked and guarded.